src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,116 @@
+                PreparedStatement getManufacturerStatement
+                        = connection.prepareStatement(getManufacturer)) {
+            getManufacturerStatement.setObject(1, id);
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(createManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException("Cannot get manufactured from DB by ID " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String selectAllQuery = "SELECT * FROM manufacturers "
+                + "WHERE is_deleted = false;";
+        List<Manufacturer> manufacturerList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getManufacturerStatement
+                        = connection.prepareStatement((selectAllQuery))) {
+            ResultSet resultSet = getManufacturerStatement.executeQuery();
+            while (resultSet.next()) {
+                manufacturerList.add(createManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot get all manufactures from DB", e);
+        }
+        return manufacturerList;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateManufacture = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE is_deleted = false AND id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement
+                        = connection.prepareStatement((updateManufacture))) {
+            getManufacturerStatement.setString(1, manufacturer.getName());
+            getManufacturerStatement.setString(2, manufacturer.getCountry());
+            getManufacturerStatement.setObject(3, manufacturer.getId());
+            getManufacturerStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot update manufacturer: " + manufacturer, e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getManufacturerStatement
+                        = connection.prepareStatement((deleteRequest))) {
+            getManufacturerStatement.setLong(1, id);
+            return getManufacturerStatement.executeUpdate() >= 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cannot delete manudacturer from DB by id: " + id, e);
+        }
+    }
+
+    private Manufacturer createManufacturer(ResultSet resultSet) {
```

Comment:
```suggestion
    private Manufacturer createManufacturer(ResultSet resultSet) throws SQLException {
```
let's throw exception further and handle it in crud methods